src/AdoNetCore.AseClient/Internal/ValueReader.cs

```diff
@@ -32,6 +33,7 @@ public static object Read(Stream stream, FormatItem format, Encoding enc)
                 case TdsDataType.TDS_INTN:
                     switch (stream.ReadByte())
                     {
+                        case 0: return DBNull.Value;
```

Comment:
There's a catch-all `return null` at the end of the function, I reckon it should be changed too.

Reply:
Saw that, wasn't sure if it was a catch-all in the sense that stuff wasn't implemented, or if it was a catch-all where we know all the cases that should reach it?

If the former, then throwing would be better, if the latter, then DBNull.Value would be appropriate.

Reply:
Intent was the latter (switch `default` should handle unsupported cases), but in practise it could have ended up being the former 😄

Reply:
Ok, I've fixed this as you suggested.